packages/ui/src/models/visualization/flows/kamelet-visual-entity.ts

```diff
@@ -28,17 +31,29 @@ export class KameletVisualEntity extends AbstractCamelVisualEntity {
 
   getComponentSchema(path?: string | undefined): VisualComponentSchema | undefined {
     if (path === ROOT_PATH) {
-      /** A better schema will be provided at a later stage */
       return {
         title: 'Kamelet',
-        schema: {} as KaotoSchemaDefinition['schema'],
+        schema: this.getRootKameletSchema(),
         definition: this.route,
       };
     }
 
     return super.getComponentSchema(path);
   }
 
+  private getRootKameletSchema(): KaotoSchemaDefinition['schema'] {
+    const processorDefinition = CamelCatalogService.getComponent(CatalogKind.Entity, 'KameletConfiguration');
+
+    if (processorDefinition === undefined) return {} as unknown as KaotoSchemaDefinition['schema'];
```

Comment:
This type should be `Record<string, unknown>` instead since it's not a schema but rather the model from the source code. 😃

Reply:
Actually, my bad, this is not the sourcecode but rather a catalog entry, so the type should be the result of the `getComponent()` method